marathon-bigip-ctlr.py

```diff
@@ -600,53 +750,84 @@ def create_config_marathon(bigip, apps):
         # The Marathon appId contains the full path, replace all '/' in
         # the name with '_'
         frontend_name = frontend_name.replace('/', '_')
-        f5_service['name'] = frontend_name
+
         if app.bindAddr:
             logger.debug("Frontend at %s:%d with backend %s", app.bindAddr,
                          app.servicePort, backend)
 
+        # pool members
+        members = []
+        key_func = attrgetter('host', 'port')
+        for backendServer in sorted(app.backends, key=key_func):
+            logger.debug("Found backend server at %s:%d for app %s",
+                         backendServer.host, backendServer.port, app.appId)
+
+            # Resolve backendServer hostname to IP address
+            ipv4 = resolve_ip(backendServer.host)
+
+            if ipv4 is not None:
+                member = {
+                    'address': ipv4,
+                    'port': backendServer.port,
+                    'state': 'user-up',
+                    'session': 'user-enabled'
+                }
+                members.append(member)
+            else:
+                logger.warning("Could not resolve ip for host %s, "
+                               "ignoring this backend", backendServer.host)
+
         if app.iapp:
             # Translate from the internal properties we set on app to the
-            # naming expected by the common f5_service['iapp']
+            # naming expected by the iapp.
             # Only set properties that are actually present.
-            #
-            # tableName would be better as poolMemberTableName but it is
-            # required to be tableName to match the tableName produced by
-            # the k8s-bigip-ctlr
-            f5_service['iapp'] = {}
+            cfg = {}
             for k, v in {'template': 'iapp',
                          'tableName': 'iappPoolMemberTableName',
                          'poolMemberTable': 'iappPoolMemberTable',
                          'tables': 'iappTables',
                          'variables': 'iappVariables',
                          'options': 'iappOptions'}.iteritems():
                 if hasattr(app, v):
-                    f5_service['iapp'][k] = getattr(app, v)
+                    cfg[k] = getattr(app, v)
 
             # Decode the tables
             for key in app.iappTables:
-                f5_service['iapp']['tables'][key] = \
+                cfg['tables'][key] = \
                     json.loads(app.iappTables[key])
```

Comment:
Probably need to catch this - it's user input that could be invalid JSON.  Need to not blow up, and need to provide a good error message.